### ISSUE TITLE
BOOST detection on Windows is not compatible with prebuilt binaries, ABI tag is wrong

### ISSUE THREAD

**Issue by mqudsi** [mesonbuild/meson#4044]
The ABI tag for the pre-built Windows BOOST binaries includes the system architecture:

```
C:\Users\mqudsi>fd boost_system C:\local
C:\local\boost_1_68_0\lib64-msvc-14.1\boost_system-vc141-mt-gd-x64-1_68.dll
C:\local\boost_1_68_0\lib64-msvc-14.1\boost_system-vc141-mt-gd-x64-1_68.lib
C:\local\boost_1_68_0\lib64-msvc-14.1\boost_system-vc141-mt-x64-1_68.dll
C:\local\boost_1_68_0\lib64-msvc-14.1\boost_system-vc141-mt-x64-1_68.lib
C:\local\boost_1_68_0\lib64-msvc-14.1\libboost_system-vc141-mt-gd-x64-1_68.lib
C:\local\boost_1_68_0\lib64-msvc-14.1\libboost_system-vc141-mt-s-x64-1_68.lib
C:\local\boost_1_68_0\lib64-msvc-14.1\libboost_system-vc141-mt-sgd-x64-1_68.lib
C:\local\boost_1_68_0\lib64-msvc-14.1\libboost_system-vc141-mt-x64-1_68.lib
```

However, meson searches for the library without the `-x64` modifier in the ABI tag:

```log

Couldn't find library 'boost_system-vc141-mt-gd-1_68' for boost module 'system'  (ABI tag = '-vc141-mt-gd-1_68')
Couldn't find library 'boost_system-mt' for boost module 'system'  (ABI tag = '-mt')
Requested Boost library 'system' not found
Boost library directory is C:\local\boost_1_68_0\lib64-msvc-14.1
Dependency Boost (system) found: NO
```

The libraries were installed from the prebuilt binaries available at https://sourceforge.net/projects/boost/files/boost-binaries/1.68.0/

**Comment by bredelings** [comment]
Actually, I think this is fixed in git master, so version 0.48 should have the fix.
https://github.com/mesonbuild/meson/commit/448b93fbe57bff6f2c4708b482ec352c2ed37793
It also looks like @nirbheek tagged it for 0.47.2

**Comment by mqudsi** [comment]
Yup, working with 0.48.0.dev1.

Thanks, @bredelings and @nirbheek
